services/web/server/tests/unit/with_dbs/03/test_trash.py

```diff
@@ -174,3 +176,222 @@ async def test_trash_projects(  # noqa: PLR0915
+    await assert_status(resp, status.HTTP_200_OK)
+
+    page = Page[FolderGet].parse_obj(await resp.json())
+
+    assert page.meta.total == 1
+    assert page.data[0].folder_id == folder.folder_id
+
+    # UNTRASH
+    assert client.app.router["untrash_folder"].url_for(folder_id="folder_id") == URL(
+        "/v0/folders/folder_id:untrash"
+    )
+
+    resp = await client.post(f"/v0/folders/{folder.folder_id}:untrash")
+    data, _ = await assert_status(resp, status.HTTP_204_NO_CONTENT)
+
+    # GET
+    resp = await client.get(f"/v0/folders/{folder.folder_id}")
+    data, _ = await assert_status(resp, status.HTTP_200_OK)
+    got = FolderGet.parse_obj(data)
+
+    assert got.folder_id == folder.folder_id
+    assert got.trashed_at is None
+
+
+@pytest.mark.acceptance_test(
+    "For https://github.com/ITISFoundation/osparc-simcore/pull/6642"
+)
+async def test_trash_folder_with_content(
+    client: TestClient,
+    logged_user: UserInfoDict,
+    user_project: ProjectDict,
+    mocked_catalog: None,
+    mocked_director_v2: None,
+):
+    assert client.app
+    project_uuid = UUID(user_project["uuid"])
+
+    # CREATE a folder
+    resp = await client.post("/v0/folders", json={"name": "My first folder"})
+    data, _ = await assert_status(resp, status.HTTP_201_CREATED)
+    folder = FolderGet.parse_obj(data)
+
+    # CREATE a SUB-folder
+    resp = await client.post(
+        "/v0/folders",
+        json={"name": "My subfolder 1", "parentFolderId": folder.folder_id},
+    )
+    data, _ = await assert_status(resp, status.HTTP_201_CREATED)
+    subfolder = FolderGet.parse_obj(data)
+
+    # MOVE project to SUB-folder
+    resp = await client.put(
+        f"/v0/projects/{project_uuid}/folders/{subfolder.folder_id}"
+    )
+    await assert_status(resp, status.HTTP_204_NO_CONTENT)
+
+    # CHECK created
+    resp = await client.get("/v0/folders")
+    await assert_status(resp, status.HTTP_200_OK)
+    page = Page[FolderGet].parse_obj(await resp.json())
+    assert page.meta.total == 1
+    assert page.data[0] == folder
+
+    resp = await client.get("/v0/folders", params={"folder_id": f"{folder.folder_id}"})
+    await assert_status(resp, status.HTTP_200_OK)
+    page = Page[FolderGet].parse_obj(await resp.json())
+    assert page.meta.total == 1
+    assert page.data[0] == subfolder
+
+    resp = await client.get(
+        "/v0/projects", params={"folder_id": f"{subfolder.folder_id}"}
+    )
+    await assert_status(resp, status.HTTP_200_OK)
+    page = Page[ProjectListItem].parse_obj(await resp.json())
+    assert page.meta.total == 1
+    assert page.data[0].uuid == project_uuid
+    assert page.data[0].folder_id == subfolder.folder_id
+
+    # ---------------------------------------------------------------------
+
+    # TRASH folder
+    resp = await client.post(f"/v0/folders/{folder.folder_id}:trash")
+    await assert_status(resp, status.HTTP_204_NO_CONTENT)
+
```

Comment:
do I get it right that if there are more than 100 children in the trashed folder it will raise an exception?
and what is the explicit non-explicit thing bringing?

Reply:
See my explanation above. It is to keep a check on a temporary solution. I prefer to have a `ValueError` with information (and now with the troubleshooting is super clear!) than the database blowing up uncontrollably :-)

Reply:
ok

Reply:
At the end i removed that ValueError. will address this problem using bulk updates